xbmc/addons/AddonRepos.h

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "Addon.h"
+
+#include <map>
+#include <vector>
+
+namespace ADDON
+{
+/**
+ * Struct - RepoInfo
+ */
+struct RepoInfo
+{
+  std::string m_repoId;
+  std::string m_origin;
+};
+
+class CAddonMgr;
+
+/**
+ * Class - CAddonRepos
+ */
+class CAddonRepos
+{
+public:
+  CAddonRepos(const CAddonMgr& addonMgr_) : addonMgr(addonMgr_){};
+
+  void SetupFullAddonsByRepoMap(const VECADDONS& addons);
+  void BuildUpdateList(const VECADDONS& installed, VECADDONS& updates) const;
+
+  /*!
+   * Checks if the origin-repository of a given addon is defined as official repo
+   * and verify if the origin-path is also defined and matching
+   * \param addon pointer to addon to be checked
+   * \param bCheckAddonPath also check origin path
+   */
+  bool IsFromOfficialRepo(const AddonPtr& addon, bool bCheckAddonPath) const;
+
+  const std::map<std::string, AddonPtr>& GetLatestPrivateVersions() const;
+  const std::map<std::string, AddonPtr>& GetLatestOfficialVersions() const;
+  const std::map<std::string, std::map<std::string, AddonPtr>>& GetLatestVersionsByRepo() const;
+
+private:
+  const CAddonMgr& addonMgr;
```

Comment:
Move this down to same place as other member vars. Good rule of thumb is functions first, then vars.